receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go

```diff
@@ -63,6 +64,13 @@ func (s *scraper) scrape(_ context.Context) (pdata.MetricSlice, error) {
 		return metrics, scrapererror.NewPartialScrapeError(err, metricsLen)
 	}
 
+	if s.config.PerCPU {
+		divisor := float64(runtime.NumCPU())
```

Comment:
Do we better expose "num_cpu" as an individual metric, and if people are interested in this they can use the backend of choice, or a "processor" to compute this. What do you think?

Reply:
Hi @bogdandrutu this is a legitimate but difficult question ^^
The best I can do is to explain the full goal of this PR and let maintainers of this project to decide.

## context

We are customer of former SignalFx product aquired by Splunk more then 2 years ago.
We are trying to migration from the [deprecated smart agent](https://github.com/signalfx/signalfx-agent/blob/main/docs/smartagent-deprecation-notice.md) to OpenTelemetry Collector.

In my opinion this deprecation by Splunk was a bit rushed and let their customers with a migration difficult to achieve given that the lack of documentation for SignalFx / Splunk specific requirements and common usages (which is not your problem!).

This PR is a tiny fragment of the work I am doing to make this migration seamless and transparent.

## goal

About the load metrics, we come from the smart agent load monitor: https://docs.signalfx.com/en/latest/integrations/agent/monitors/load.html which provide an option to average the load metrics per cpu number.

This PR aims to bring a similar feature for parity between signalfx smart agent and open telemetry collector to be able to use the otel receiver but keep all existing dependent resources working like charts, detectors..

## workaround

For now, as workaround I still use the smart agent monitor instead of hostmetrics receiver:

```yaml
receivers:
  hostmetrics:
    collection_interval: 10s
    scrapers:
      cpu:
      disk:
      filesystem:
      memory:
      network:
      #load:
      paging:
      processes:
  smartagent/load:
    type: load
    perCPU: true
```

but I would prefer to drop smart agent monitors in profit of otel receivers if possible

## your suggestion

your suggestion is full of sens obviously and in fact I could already create the average per cpu on load outside the receiver because I already have everything I need.

Indeed, the SignalFx exporter for Otel Collector already expose a metric for number of cpu here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4f91eb07feb5cd75c5739a472fc9f2c38ec5295f/exporter/signalfxexporter/internal/translation/constants.go#L123

so I can basically divide the load metric from hostmetrics receiver by this `cpu.num_processors` and I will get similar value I had with smart agent load monitor.

## the problem

doing this way will force users to update all existing resources to use a more complex query to calculate this load averaged per cpu.

the problem is we have lot of customers on different signalfx organizations, we do not manage every resources ourselves and without to speak about the complexity of tracking, detect and update resources out of our scope this is also tricky due to contractual / responsibility / permission considerations.

even for the resources we manage ourselves properly (iac, git, terraform etc) and are of our responsibility it can be tedious to update existing resources without insert new source of mistakes. For example, our detector to create alert on load metrics is very simple: https://github.com/claranet/terraform-signalfx-detectors/blob/master/modules/smart-agent_system-common/detectors-gen.tf#L80
We explicitly instruct users of this module to enable `perCPU:true` option and everything is good.
We could change this module to modify the query and make the division with `cpu.num_processors`. In this way, the detector will be compatible with both otel collector and smart agent.
Nevertheless, if I do this I know some users will upgrade to the new version of the module without to remove the `perCPU` option from the agent configuration and so will break their detector silently ...

## conclusion

It is a mess !

Honestly I fully understand if you do not want to integrate a "useless" feature in otel collector especially if it is only for a specific vendor.

but here is the full explanation, this seems to me to be the less disruptive, the safest and the most straightforward way to handle my need.

I also would like to argue that:
- this option still can be useful to others (no splunk customers)
- this could ease the setup of monitoring in general
- this way will result in less required metrics (1 instead of 2) which can be important as much for vendor like splunk where each metric can increase the price as for opensource based solution where storage of metrics remains the most difficult part

ok not sure of the relevance of these arguments but nothing ventured, nothing gained :)